velox/functions/prestosql/URLFunctions.h

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/regex.hpp>
+#include <cctype>
+#include "folly/Uri.h"
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+class UrlParser {
+ protected:
+  StringView submatch(const boost::cmatch& match, int idx);
+
+  template <typename TInString>
```

Comment:
Does this need to be a template? Do you ever use any type other than StringView?

Reply:
Other string functions use the same template - so for sake of consistency I was using this. I can change it to StringView though.

Reply:
For the sake of simplicity, let's not use templates when there is only one template type possible.